spec/core/instances_spec.rb

```diff
@@ -482,11 +559,48 @@
   end
 
   describe '#internal_values_of' do
-    it 'returns the internal values of the given values' do
-      values          = [:sunny, 85, 85, :FALSE, :no]
-      internal_values = [0, 85.0, 85.0, 1, 1]
 
-      expect(subject.internal_values_of(values)).to eq internal_values
+    context 'when passing an array' do
+      it 'returns an array of internal values of the given values' do
+        values = [:sunny, 85, 85, :FALSE, :no]
+        internal_values = [0, 85.0, 85.0, 1, 1]
+
+        expect(subject.internal_values_of(values)).to eq internal_values
+      end
+    end
+
+    context 'when passing a hash' do
+      values = {
+        outlook: :sunny,
+        temperature: 85,
+        humidity: 85,
+        windy: :FALSE,
+        play: :no
+      }
+
+      internal_values = {
+        outlook: 0,
+        temperature: 85.0,
+        humidity: 85.0,
+        windy: 1,
+        play: 1
+      }
```

Comment:
I figure `values` & `internal_values` should be `let`-blocks. Doesn't make a difference for the static values here, but just for consistency.